src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -22,6 +22,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <_LinkSemaphore>$(IntermediateOutputPath)Link.semaphore</_LinkSemaphore>
   </PropertyGroup>
 
+  <!-- We disable startup hooks for trimmed apps here so that the feature
+       switch can flow to the runtimeconfig.json. Startup hooks are disabled
+       by default since they may require assemblies, types or members that
+       could be removed by the linker, causing a trimmed app to crash. -->
+  <PropertyGroup Condition="'$(EnableUnsafeBinaryFormatterSerialization)' == '' And
```

Comment:
Why is this conditioned on binary formatter serialization? Could you add a comment?

Reply:
I guess this is meant to be conditioned on `StartupHookSupport`

Reply:
My bad 😄 - fixed